Engine/DataFeeds/TextSubscriptionDataSourceReader.cs

```diff
@@ -34,7 +39,16 @@ public class TextSubscriptionDataSourceReader : ISubscriptionDataSourceReader
         private readonly BaseData _factory;
         private readonly DateTime _date;
         private readonly SubscriptionDataConfig _config;
+        private bool _shouldCacheDataPoints;
         private readonly IDataCacheProvider _dataCacheProvider;
+        private static readonly MemoryCache BaseDataSourceCache = new MemoryCache("BaseDataSourceCache",
+            // Cache can use up to 70% of the installed physical memory
+            new NameValueCollection{ { "physicalMemoryLimitPercentage", "70"} });
+        private static readonly CacheItemPolicy CachePolicy = new CacheItemPolicy
+        {
+            // Cache entry should be evicted if it has not been accessed in given span of time:
+            SlidingExpiration = TimeSpan.FromMinutes(5)
```

Comment:
LRU instead of time-based caching might make more sense here.

Reply:
Added this sliding expiration policy to avoid reaching the memory limit in a long running algorithm. I believe LRU will be used when the memory limit is reached.